### ISSUE TITLE
Extension api: Add support to the runQuery command to pass custom sql

### ISSUE THREAD

**Issue by bvogelzang** [microsoft/vscode-mssql#424]
- MSSQL Extension Version: 0.1.5
- VSCode Version: 1.7.2
- OS Version: Mac OS X 10.11.6 (El Capitan)

I would love to write my own extension to run sql snippets and queries. However, at the moment all I can do is rely on the currently highlighted/selected code inside the text editor. I don't see any way to pass a custom sql string to the runQuery command.

Could we add support to the runQuery command to pass custom sql or add another method to enable this?

I'd love to be able to do something like the following inside of a custom extension:
```javascript
// get current text selection
var selection = editor.selection;
var text = editor.document.getText(selection);

// import mssql api
var mssql = vscode.extensions.getExtension('ms-mssql.mssql');
var importedApi = mssql.exports;

// combine sql snippet with selected text and run 
importedApi.runQuery('select top 100 * from ' + text);
```

**Comment by kevcunnane** [comment]
This is a good suggestion and could enable scenarios like allowing a user to execute SQL embedded in .cs or .ts files for testing of  testing of simple SQL queries embedded in other languages.

@erickangMSFT - I can see this being something that needs a security review before implementation given previous discussions on this matter. We should sync on this to see what is feasible - for instance we could prompt a user once to allow access to other extensions and from then on allow use of the API?

**Comment by buzzzzer** [comment]
Hi. Greate feature.
It is planned for implementation ?

**Comment by 4mitch** [comment]
I'm also eager for this implementation. I even not so want to make custom extentions or tasks as just want to add custom SQL Server command to context menu based on selected text. I.e. run SP_HELP on selected table, proc name etc, or run other sp_procs.

**Comment by dylan-azucena** [comment]
Yeah this would create the ability to mimic management studio's Query Shortcuts. In VSCode I currently have a keyboard shortcut for adding text like sp_help to a selection, but it would be nice to run it all in one keystroke.

**Comment by Ivacsu** [comment]
Hi , The functionality that @bvogelzang  asked for is already available ?  ,  Or is there another way to do that ? ,

I want to do .
"Ctrl + 3 " : SELECT * FROM 

 **SSMS**
![image](https://user-images.githubusercontent.com/14947343/98970313-c8ca3700-24dd-11eb-9cd4-8a59f965a25b.png)

**Azure Data Studio** 

![image](https://user-images.githubusercontent.com/14947343/98970528-06c75b00-24de-11eb-82fe-c003d3346e5c.png)

Thank

**Comment by Mjongeling07** [comment]
Now with the retirement of ADS, many users will transition to VS Code. I feel this really needs to become part of the extension. Having shortcuts like in SSMS would be fantastic!

https://learn.microsoft.com/en-us/azure-data-studio/whats-happening-azure-data-studio

**Comment by kburtram** [comment]
Closing this issue since we currently support a suitable workaround:

Another extension can insert custom SQL into an open SQL editor pane and invoke our existing mssql.runQuery command. This ensures users see exactly what's being executed, maintaining transparency and implicitly addressing security concerns without the complexity of additional permission UIs.

Given the availability of this transparent approach and potential security complexities with direct API execution, we won't implement the original request at this time. Thanks everyone for the discussion!

**Comment by carloslongen** [comment]
Please, do we have any example of how to accomplish this **suitable workaround** ?
